src/compile/mark/valueref.ts

```diff
@@ -83,18 +75,22 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
+      if (scale.type === 'bin-linear') {
+        return {
+          signal: `(scale("${scaleName}", ${field(channelDef, {binSuffix: 'start', datum: true})}) + scale("${scaleName}", ${field(channelDef, {binSuffix: 'end', datum: true})}))/2`
```

Comment:
> bigger fish to fry

Yes, but this takes 1 min to fix and I didn't know why you spend a minute earlier to make this part worse :p